src/api/java/com/minecolonies/api/colony/colonyEvents/IColonyEvent.java

```diff
@@ -0,0 +1,142 @@
+package com.minecolonies.api.colony.colonyEvents;
+
+import com.minecolonies.api.colony.IColony;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Interface for colony event types.
+ */
+public interface IColonyEvent
```

Comment:
IColonyEvent is named a bit weirdly, especially since you force it to have a spawnpoint.
Maybe think of something that more allines with your goals.

Reply:
Something like `IColonyEntitySpawnEvent` or `IColonyGroupSpawnEvent` possibly with some abstraction into an `IColonySpawnEvent` and `IColonyEvent`

Reply:
Also you might want to have an interface that describes an event that spawns a structure.
So that raid events are both spawning and have structure handling.

This makes expanding this system easier in the future -> Think of camps of traiders that spawn outside the city walls to trade with etc.

Reply:
ye you're right indeed, colony event doesnt necessarily need entities, splitting it makes sense. An interface for spawning structures alongside an event should be useful too

Reply:
I've seperated them into Colony/Spawn/entitySpawn/raid/structureSpawn types now, think that should fit it